xcm/xcm-executor/src/lib.rs

```diff
@@ -320,6 +320,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				// We must do this first in order to resolve wildcards.
 				let assets = holding.saturating_take(assets);
 				for asset in assets.assets_iter() {
+					ensure!(
+						Config::IsTeleporter::filter_asset_location(&asset, &dest),
+						XcmError::UntrustedTeleportLocation
+					);
```

Comment:
```suggestion
```
Should remove this. Gav clarified that this filter is explicitly left out because it is non-critical (security-wise) for a chain to teleport assets **out to** (instead of **in from**) an untrusted location and allows for one-sided trust if that is desired (e.g. in a transition period where mutual trust is being established).